### PR TITLE
HY-1872: Pan Items After Scrolling in !flow Mode

### PR DESCRIPTION
**Pull request by timmccall-wf**
Problem
-------

Items are not repositioned correctly after zooming out and swiping to an adjacent item in `peek` mode, or after changing the current item in `single` mode. The zoom state is restored, but the position is left in an odd state. For example: zoom out on item 1, swipe to item two, then swipe back to item 1 and note that item 1 will be centered in the viewport, rather than have the bottom of the item aligned with the bottom of the viewport. This hurts continuity as you navigate among pages.

Solution
--------

Make sure to pan after scrolling to an item in `peek` or `single` modes.

Unit Tests
----------

Updated.

How To +10/QA
-------------

- Open the ScrollList demo with `fit=width&scroll=peek`. Ensure you have vertical overflow.
- Navigate to page 3 and zoom out to 50% (get the whole page in the viewport)
- Swipe by clicking and dragging until the previous/next item is in view
- Pull the previous page into view
- Should see the page aligned properly with the top (for following page) or bottom (for previous page) of the viewport.

@todbachman-wf

### CODE REVIEW

**Comment by rmconsole-wf** [comment]
When this pull is merged I will use the following information:
		Version: wf-uicomponents 1.1.0
		Release Ticket: HY-1862

***
*Last updated on Tuesday, February 03 04:35 PM CST*

File: examples/scroll_list/scroll-list.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ define(function(require) {
         minNumberOfVirtualItems: minNumberOfVirtualItems,
         mode: scrollMode,
         padding: 1,
-        persistZoom: zoomMode,
+        persistZoom: persistZoom,
```
Comment (tomconnell-wf) [inline examples/scroll_list/scroll-list.js]: :clap:

File: src/scroll_list/ScrollList.js [hunk 2]
```diff
@@ -1003,9 +1003,28 @@ define(function(require) {
...
+                                transformY = viewportHeight - itemHeight;
+                            }
+                            // For items after the current one, make sure they
+                            // are panned to the top of the viewport.
+                            else if (i > currentItemIndex && itemHeight > viewportHeight) {
+                                transformY = 0;
+                            }
                             if (i !== currentItemIndex) {
-                                self.getItemMap(i).zoomTo({ scale: 1 });
+                                itemMap.zoomTo({ scale: 1 });
```
Comment (tomconnell-wf) [inline src/scroll_list/ScrollList.js]: Do we need to check persistZoom here?

Comment (timmccall-wf) [inline src/scroll_list/ScrollList.js]: Line 998

Comment (tomconnell-wf) [inline src/scroll_list/ScrollList.js]: Oh, I get it, nvm.  This is for not the current page.

**Comment by tomconnell-wf** [comment]
+1

**Comment by rmconsole3-wf** [comment]
@timmccall-wf This pull request has merge conflicts, please resolve.

**Comment by timmccall-wf** [comment]
@rmconsole-wf

**Comment by todbachman-wf** [comment]
+10. Despite my best attempts, I couldn't break it. I tested in the ScrollList demo as well as HVR with zoomCanvas.

**Comment by charliekump-wf** [comment]
QA Resource Approval: +1

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Unit test created/updated  
- [x] All unit tests pass 

Merging into master.

**Pull request merged**